### PR TITLE
fix dbt asset checks on sources

### PR DESCRIPTION
**Pull request by johannkm**
https://github.com/dagster-io/dagster/pull/19319 calls `manifest["nodes"][<source id>]`, instead use the `dbt_nodes` dict that has nodes and sources and more.

Test plan: ran this against a failing manifest.json and this fixed the issue. Adding a source as an additional test dep to the test project seems like a pain but could do it

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #19639** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19639?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19639?utm_source=stack-comment).

**Review by rexledesma — DISMISSED** [review]
Let's add that test you mentioned 🙏

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/seeds/raw_cereals.csv
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/seeds/raw_cereals.csv:1]: Do we need to introduce this file? Can't we just label `raw_customers` as a source? And then add that as a relationship to `customer_id` on `customers?

Comment (rexledesma) [reply python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/seeds/raw_cereals.csv:1]: Added this in https://github.com/dagster-io/dagster/pull/19639/commits/cbae8ee05c6635ad39bb736d3899f4ed6974143a

**Comment by johannkm** [comment]
lgtm

**Review by rexledesma — APPROVED** [review]

**Pull request merged**